sql/core/src/test/scala/org/apache/spark/sql/SubquerySuite.scala

```diff
@@ -2204,4 +2204,20 @@ class SubquerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
             |""".stripMargin),
       Row("2022-06-01"))
   }
+
+  test("SPARK-39511: Push limit 1 to right side if join type is Left Semi/Anti") {
```

Comment:
this enhancement should also be mentioned in the PR description. Now we apply this optimization to left/semi join generated from subquereis as well.

Reply:
Updated  PR description.